packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -20,10 +20,20 @@ const Error = Loadable({
 
 const Routes = observer(({ history, location, passthrough }) => {
     const { client, common } = useStore();
-    const { is_logged_in, is_logging_in } = client;
+    const { is_authorize, is_logged_in, is_logging_in } = client;
     const { error, has_error, setAppRouterHistory, addRouteHistoryItem, setInitialRouteHistoryItem } = common;
     const initial_route = React.useRef(null);
     const unlisten_to_change = React.useRef(null);
+    const {
+        subscribe,
+        rest: { isSubscribed },
+    } = useP2PSettings();
+
+    React.useEffect(() => {
+        if (is_authorize && !isSubscribed) {
+            subscribe();
+        }
+    }, [is_authorize, isSubscribed, subscribe]);
```

Comment:
I need to call subscribe in here, cashier and in p2p, because if i don't, for some reason the data from p2p_settings is not passed to these components despite being wrapped with the P2PSettingsProvider.